crates/modelardb_server/src/storage/data_transfer.rs

```diff
@@ -160,14 +170,67 @@ impl DataTransfer {
     /// If the value is changed successfully return [`Ok`], otherwise return [`ParquetError`].
     pub(super) async fn set_transfer_batch_size_in_bytes(
         &mut self,
-        new_value: usize,
+        new_value: Option<usize>,
     ) -> Result<(), ParquetError> {
-        self.transfer_larger_than_threshold(new_value).await?;
+        if let Some(new_batch_size) = new_value {
+            self.transfer_larger_than_threshold(new_batch_size).await?;
+        }
+
         self.transfer_batch_size_in_bytes = new_value;
 
         Ok(())
     }
 
+    /// If a new transfer time is given, stop the existing task transferring data periodically,
+    /// if there is one, and start a new task. If `new_value` is [`None`], the task is just stopped.
+    /// `data_transfer` is needed as an argument instead of using `self` so it can be moved into the
+    /// periodic task.
+    pub(super) fn set_transfer_time_in_seconds(
+        &mut self,
+        new_value: Option<usize>,
+        data_transfer: Arc<RwLock<Option<Self>>>,
+    ) {
+        // Stop the current task periodically transferring data if there is one.
+        if let Some(task) = &self.transfer_scheduler_handle {
+            task.abort();
+        }
+
+        // If a transfer time was given, create the scheduler that periodically transfers data.
+        self.transfer_scheduler_handle = if let Some(transfer_time) = new_value {
+            let mut scheduler = AsyncScheduler::new();
+            scheduler
+                .every((transfer_time as u32).seconds())
+                .run(move || {
+                    // Clone the Arc so only the clone is moved.
+                    let data_transfer = data_transfer.clone();
+
+                    async move {
+                        // unwrap() is safe as this code can only be reached if data_transfer is some.
+                        data_transfer
+                            .write()
+                            .await
+                            .as_ref()
+                            .unwrap()
+                            .transfer_larger_than_threshold(0)
+                            .await
+                            .expect("Periodic data transfer failed.");
+                    }
+                });
+
+            // Start a tokio task that checks if the scheduler has pending tasks and runs them if so.
+            let join_handle = tokio::spawn(async move {
+                loop {
+                    scheduler.run_pending().await;
+                    tokio::time::sleep(Duration::from_millis(1000)).await;
```

Comment:
Could we remove this extra sleep so when the data is transferred only depends on transfer time in seconds instead of two timers and also remove `clockwerk` by simply creating a [`tokio::task`](https://docs.rs/tokio/latest/tokio/task/index.html) that sleeps for `transfer_time_in_seconds` and calls `transfer_larger_than_threshold(0)`? Since that would also create a [`JoinHandle`](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html) it should still be possible to use `task.abort()` when updating transfer time in seconds.

Reply:
Good idea, the idea of using a crate was mainly due to initially wanting the scheduling to be more advanced (fx. cron) but since we use fairly simply scheduling it seems fine to use just tokio.